x-pack/plugins/maps/public/components/widget_overlay/layer_control/layer_toc/toc_entry/view.js

```diff
@@ -112,33 +125,69 @@ export class TOCEntry extends React.Component {
     );
   }
 
-  _renderLayerName() {
-    const displayName = (
-      <div style={{ width: 180 }} className="eui-textTruncate eui-textLeft">
-        {this.state.displayName}
-      </div>
-    );
-
+  _renderLayerIcons() {
     if (this.props.isReadOnly) {
-      return (
-        <EuiText>
-          {displayName}
-        </EuiText>
-      );
+      return null;
     }
 
     return (
-      <EuiLink
-        color="text"
-        onClick={this._openLayerPanelWithCheck}
-        data-test-subj={
-          `mapOpenLayerButton${this.state.displayName
-            ? this.state.displayName.replace(' ', '_')
-            : ''}`
-        }
-      >
-        {displayName}
-      </EuiLink>
+      <div className="mapTocEntry__layerIcons">
+
+        <EuiButtonIcon
+          iconType="pencil"
+          aria-label={i18n.translate('xpack.maps.layerControl.tocEntry.editButtonAriaLabel', {
+            defaultMessage: 'Edit layer'
+          })}
+          title={i18n.translate('xpack.maps.layerControl.tocEntry.editButtonTitle', {
+            defaultMessage: 'Edit layer'
+          })}
+          onClick={this._openLayerPanelWithCheck}
+        />
+
+        <EuiButtonIcon
+          iconType="grab"
+          color="subdued"
+          title={i18n.translate('xpack.maps.layerControl.tocEntry.grabButtonTitle', {
+            defaultMessage: 'Reorder layer'
+          })}
+          aria-label={i18n.translate('xpack.maps.layerControl.tocEntry.grabButtonAriaLabel', {
+            defaultMessage: 'Reorder layer'
+          })}
+          className="mapTocEntry__grab"
+          {...this.props.dragHandleProps}
+        />
+
+      </div>
+    );
+  }
+
+  _renderDetailsToggle() {
+    const { isLayerDetailsOpen } = this.props;
+    return (
+      <span className="mapTocEntry__detailsToggle">
+        <button
+          className="mapTocEntry__detailsToggleButton"
+          aria-label={isLayerDetailsOpen
+            ? i18n.translate('xpack.maps.layerControl.tocEntry.hideDetailsButtonAriaLabel', {
+              defaultMessage: 'Hide layer details'
+            })
+            : i18n.translate('xpack.maps.layerControl.tocEntry.showDetailsButtonAriaLabel', {
+              defaultMessage: 'Show layer details'
+            })
+          }
+          title={isLayerDetailsOpen
+            ? i18n.translate('xpack.maps.layerControl.tocEntry.hideDetailsButtonTitle', {
+              defaultMessage: 'Hide layer details'
+            })
+            : i18n.translate('xpack.maps.layerControl.tocEntry.showDetailsButtonTitle', {
+              defaultMessage: 'Show layer details'
+            })
+          }
+          onClick={this._toggleLayerDetailsVisibility}
+        >
+          <EuiIcon className="eui-alignBaseline" type={isLayerDetailsOpen ? 'arrowUp' : 'arrowDown'} size="s" />
```

Comment:
Any thoughts on _not_ showing the arrow when the contents of the layer-details is empty? That way, it doesn't show up for e.g. a raster layer of a layer with only statically styled contents.

Reply:
good idea. I added `hasLegendDetails` to layer interface. The vector layer will return true if the style has any dynamic styles